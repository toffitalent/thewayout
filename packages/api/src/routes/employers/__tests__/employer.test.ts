import { faker } from '@faker-js/faker';
import { auth, fixtures, request, resetDb } from '@test';

describe('Routes > Employers > employer', () => {
  beforeEach(async () => {
    await resetDb();

    faker.seed(1);
  });

  describe('PATCH /api/v1/employers/:id', () => {
    test('updates employer profile', async () => {
      const res = await request()
        .patch(`/api/v1/employers/${fixtures.employer1.id}`)
        .send({
          name: 'Test',
          availableJobsCount: 10,
          availableProfilesUncloak: 10,
        })
        .use(auth(fixtures.user6.id, ['employer']));

      expect(res.status).toBe(200);
      expect(res.body).toMatchSnapshot();
      // limits can be update only by admin
      expect(res.body.availableJobsCount).not.toBe(10);
      expect(res.body.availableProfilesUncloak).not.toBe(10);
    });

    test('return not found error for not authorized employer', async () => {
      const res = await request()
        .patch(`/api/v1/employers/${fixtures.employer1.id}`)
        .send({
          name: 'Test',
        })
        .use(auth(fixtures.user5.id, ['employer']));

      expect(res.status).toBe(404);
    });

    test('updates employer limits by admin', async () => {
      const res = await request()
        .patch(`/api/v1/employers/${fixtures.employer1.id}`)
        .send({
          availableJobsCount: 10,
          availableProfilesUncloak: 10,
        })
        .use(auth(fixtures.user1.id, ['admin']));

      expect(res.status).toBe(200);
      expect(res.body.availableJobsCount).toBe(10);
      expect(res.body.availableProfilesUncloak).toBe(10);
      expect(res.body).toMatchSnapshot();
    });

    test('does not allow update limit from non-admins', async () => {
      const res = await request()
        .patch(`/api/v1/employers/${fixtures.employer1.id}`)
        .send({
          availableJobsCount: 10,
          availableProfilesUncloak: 10,
        })
        .use(auth());

      expect(res.status).toBe(403);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });
  });
});
