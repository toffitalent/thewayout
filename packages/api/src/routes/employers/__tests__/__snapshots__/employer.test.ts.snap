// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes > Employers > employer PATCH /api/v1/employers/:id does not allow update limit from non-admins 1`] = `
{
  "code": "InsufficientScope",
  "message": "Insufficient scope has been authorized for this request",
  "status": 403,
}
`;

exports[`Routes > Employers > employer PATCH /api/v1/employers/:id updates employer limits by admin 1`] = `
{
  "address": "Wolff Spring",
  "availableJobsCount": 10,
  "availableProfilesUncloak": 10,
  "avatar": null,
  "city": "North Brooks",
  "createdAt": "<Date>",
  "description": "Eum eos laborum labore ipsam libero nobis. Earum maxime porro quas dicta vero. Occaecati impedit amet eaque doloribus. Veniam adipisci doloremque voluptas.",
  "id": "01817e24-0d42-763f-b7ea-8f975994f864",
  "industry": "Accounting",
  "name": "Oberbrunner, Dare and Trantow",
  "numberOfEmployees": "10000+",
  "postalCode": "04674",
  "state": "AK",
  "updatedAt": "<Date>",
  "yearsInBusiness": "0-3",
}
`;

exports[`Routes > Employers > employer PATCH /api/v1/employers/:id updates employer profile 1`] = `
{
  "address": "Wolff Spring",
  "availableJobsCount": 1,
  "availableProfilesUncloak": 3,
  "avatar": null,
  "city": "North Brooks",
  "createdAt": "<Date>",
  "description": "Eum eos laborum labore ipsam libero nobis. Earum maxime porro quas dicta vero. Occaecati impedit amet eaque doloribus. Veniam adipisci doloremque voluptas.",
  "id": "01817e24-0d42-763f-b7ea-8f975994f864",
  "industry": "Accounting",
  "name": "Test",
  "numberOfEmployees": "10000+",
  "postalCode": "04674",
  "state": "AK",
  "updatedAt": "<Date>",
  "yearsInBusiness": "0-3",
}
`;
